src/ambianic/webapp/server/config_sources.py

```diff
@@ -0,0 +1,63 @@
+
+from werkzeug.exceptions import NotFound, BadRequest
+from ambianic import config_manager
+
+source_model = {
+    "id": str,
+    "uri": str,
+    "type": str,
+    "live": bool
+}
+
+source_types = ["video", "audio", "image"]
+
+
+def validate(source_id, source):
+    """Validate input object"""
+
+    if not isinstance(source, dict):
+        raise BadRequest("source should be an object")
```

Comment:
This is a bit of a generic error message. Maybe replace with something more specific like
`Source should be a valid dictionary of pipeline source objects'.